### PR TITLE
add k8-1.0: enable linux-aarch64 build

### PR DESCRIPTION
**Pull request by guozhicong**
Describe your pull request here

----

Please read the [guidelines for Bioconda recipes](https://bioconda.github.io/contributor/guidelines.html) before opening a pull request (PR).

### General instructions

* If this PR adds or updates a recipe, use "Add" or "Update" appropriately as the first word in its title.
* New recipes not directly relevant to the biological sciences need to be submitted to the [conda-forge channel](https://conda-forge.org/docs/) instead of Bioconda.
* PRs require reviews prior to being merged. Once your PR is passing tests and ready to be merged, please issue the `@BiocondaBot please add label` command.
* Please post questions [on Gitter](https://gitter.im/bioconda/Lobby) or ping `@bioconda/core` in a comment.

### Instructions for avoiding API, ABI, and CLI breakage issues
Conda is able to record and lock (a.k.a. pin) dependency versions used at build time of other recipes.
This way, one can avoid that expectations of a downstream recipe with regards to API, ABI, or CLI are violated by later changes in the recipe.
If not already present in the meta.yaml, make sure to specify `run_exports` (see [here](https://bioconda.github.io/contributor/linting.html#missing-run-exports) for the rationale and comprehensive explanation).
Add a `run_exports` section like this:

```yaml
build:
  run_exports:
    - ...

```

with `...` being one of:

| Case                             | run_exports statement                                               |
| -------------------------------- | ------------------------------------------------------------------- |
| semantic versioning              | `{{ pin_subpackage("myrecipe", max_pin="x") }}`     |
| semantic versioning (0.x.x)      | `{{ pin_subpackage("myrecipe", max_pin="x.x") }}`   |
| known breakage in minor versions | `{{ pin_subpackage("myrecipe", max_pin="x.x") }}` (in such a case, please add a note that shortly mentions your evidence for that) |
| known breakage in patch versions | `{{ pin_subpackage("myrecipe", max_pin="x.x.x") }}` (in such a case, please add a note that shortly mentions your evidence for that) |
| calendar versioning              | `{{ pin_subpackage("myrecipe", max_pin=None) }}`    |

while replacing `"myrecipe"` with either `name` if a `name|lower` variable is defined in your recipe or with the lowercase name of the package in quotes.

### Bot commands for PR management

<details>
  <summary>Please use the following BiocondaBot commands:</summary>

Everyone has access to the following BiocondaBot commands, which can be given in a comment:

<table>
  <tr>
    <td><code>@BiocondaBot please update</code></td>
    <td>Merge the master branch into a PR.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please add label</code></td>
    <td>Add the <code>please review & merge</code> label.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please fetch artifacts</code></td>
    <td>Post links to CI-built packages/containers. <br />You can use this to test packages locally.</td>
  </tr>
</table>

Note that the <code>@BiocondaBot please merge</code> command is now depreciated. Please just squash and merge instead.

Also, the bot watches for comments from non-members that include `@bioconda/<team>` and will automatically re-post them to notify the addressed `<team>`.

</details>

### CODE REVIEW

File: recipes/k8/meta.yaml [hunk 1]
```diff
@@ -1,22 +1,48 @@
...
+  number: 0
+  run_exports:
+    - {{ pin_subpackage('k8', max_pin="x") }}
+  string: h3f4af3b_4
 requirements:
   build:
     - {{ compiler('cxx') }}
+    - _libgcc_mutex >=0.1
+    - _openmp_mutex >=4.5
+    - binutils_impl_linux-aarch64 2.40 h870a726_0
```
Comment (martin-g) [inline recipes/k8/meta.yaml:20]: This looks wrong to me. 
Why do you use `**-aarch64` dependencies here ? How is the x86_64 supposed to use them ?

**Review by martin-g — COMMENTED** [review]
You might be interested in https://github.com/attractivechaos/k8/commit/207544a4fe45300b190d455252e6e2aa53047d1e

File: recipes/k8/meta.yaml (hunk 1 above)
Comment (guozhicong) [reply recipes/k8/meta.yaml:20]: I added this tag. Doesn't it mean it 's only available in linux-aarch64's repository of conda?
image

Comment (guozhicong) [reply recipes/k8/meta.yaml:20]: @martin-g I added this tag. Doesn't it mean it 's only available in linux-aarch64's repository of conda?
image

Comment (martin-g) [reply recipes/k8/meta.yaml:20]: This is `extra`, i.e. in addition to the default platforms.
You need to use selectors, like `... # [aarch64 or osx]` to specify that something is specific for some architecture

**Comment by martin-g** [comment]
@guozhicong May I take over ? I think I see a way how to finish this task!

**Comment by guozhicong** [comment]
> @guozhicong May I take over ? I think I see a way how to finish this task!

@martin-g Of course, I dont konw how to fix it.

**Comment by martin-g** [comment]
https://github.com/bioconda/bioconda-recipes/pull/46559 - it almost works! But building Node.js from sources takes a lot of time!

**Comment by martin-g** [comment]
This PR could be closed in favour of https://github.com/bioconda/bioconda-recipes/pull/46559

**Pull request closed without merging**